pallets/parachain-staking/src/tests.rs

```diff
@@ -3582,6 +3584,47 @@ fn multiple_delegations() {
 		});
 }
 
+#[test]
+fn execute_leave_candidate_removes_delegations() {
+	ExtBuilder::default()
+		.with_balances(vec![(1, 100), (2, 100), (3, 100), (4, 100)])
+		.with_candidates(vec![(1, 20), (2, 20)])
+		.with_delegations(vec![(3, 1, 10), (3, 2, 10), (4, 1, 10), (4, 2, 10)])
+		.build()
+		.execute_with(|| {
+			// Verifies the revocation count is originally at 0
+			assert_eq!(
+				Stake::delegator_state(3)
+					.unwrap()
+					.requests
+					.revocations_count,
+				0
+			);
+
+			assert_ok!(Stake::schedule_leave_candidates(Origin::signed(2), 2));
+			assert_ok!(Stake::schedule_revoke_delegation(Origin::signed(3), 2));
+			// Verifies the revocation count has been updated to 1
+			assert_eq!(
+				Stake::delegator_state(3)
+					.unwrap()
+					.requests
+					.revocations_count,
+				1
+			);
+
+			roll_to(16);
+			assert_ok!(Stake::execute_leave_candidates(Origin::signed(2), 2));
```

Comment:
The use of 2 and 3 initially confused me, but I think I understand: 3 is a delegator, 2 is a candidate. 3 scheduled to leave, but this effectively happened when 2 left as a collator, leaving the dangling queued revoke for 3. Is that the idea?

A brief comment might help clear that up.

Reply:
> 3 is a delegator, 2 is a candidate. 3 scheduled to leave, but this effectively happened when 2 left as a collator, leaving the dangling queued revoke for 3. Is that the idea?

The test verifies that the revoke request is removed by 2's exit so there is no dangling queed revoke after 2 exits. I can add a comment above the test now.